extensions/amp-access-laterpay/0.1/amp-access-laterpay.js

```diff
@@ -30,18 +35,120 @@ export class LaterpayVendor {
   constructor(accessService) {
     /** @private @const */
     this.accessService_ = accessService;
-    // TODO: implement
+    this.win_ = this.accessService_.win;
+    this.laterpayConfig_ = this.accessService_.adapter_.getConfig();
+    this.purchaseConfig_ = null;
+    this.purchaseOptionListeners_ = [];
+    this.selectedPurchaseOption_ = null;
+    this.purchaseButton_ = null;
+
+    /** @const @private {!Xhr} */
+    this.xhr_ = xhrFor(this.win_);
   }
 
   /**
    * @return {!Promise<!JSONType>}
    */
   authorize() {
-    const win = this.accessService_.win;
-    user().assert(isExperimentOn(win, 'amp-access-laterpay'),
+    user().assert(isExperimentOn(this.win_, 'amp-access-laterpay'),
         'Enable "amp-access-laterpay" experiment');
-    // TODO: implement
-    return Promise.resolve({access: true});
+    return this.getInitialPurchaseConfig_().then(purchaseConfig => {
+      this.purchaseConfig_ = purchaseConfig;
+      if (!purchaseConfig.access) {
+        this.renderPurchaseOverlay_();
+      }
+      return {access: purchaseConfig.access};
+    });
+  }
+
+  getInitialPurchaseConfig_() {
+    const urlPromise = this.accessService_.buildUrl_(CONFIG_URL,
+        /* useAuthData */ false);
+    return urlPromise.then(url => {
+      dev().fine(TAG, 'Authorization URL: ', url);
+      return this.accessService_.timer_.timeoutPromise(
+          AUTHORIZATION_TIMEOUT,
+          this.xhr_.fetchJson(url, {
+            credentials: 'include',
+            requireAmpResponseSourceOrigin: true,
+          }));
+    });
+  }
+
+  renderPurchaseOverlay_() {
+    const laterpayList = this.win_.document.querySelectorAll(
+      'amp-access-laterpay-list')[0];
+    const listContainer = document.createElement('ul');
+    // TODO set these up somewhere else and make them configurable
+    this.purchaseConfig_.premiumcontent.title = 'Buy this article';
+    this.purchaseConfig_.premiumcontent.description =
+      'title of the article should go here';
+    listContainer.appendChild(
+      this.createPurchaseOption_(this.purchaseConfig_.premiumcontent)
+    );
+    this.purchaseConfig_.timepasses.forEach(timepass => {
+      listContainer.appendChild(this.createPurchaseOption_(timepass));
+    });
+    const purchaseButton = document.createElement('button');
+    purchaseButton.innerHTML = 'Confirm your selection';
+    purchaseButton.disabled = true;
+    this.purchaseButton_ = purchaseButton;
+    listenOnce(purchaseButton, 'click', this.handlePurchase_.bind(this));
```

Comment:
But that's not what would happen with the "login dialog" when it runs in numerous different scenarios. E.g. it can be launched as a popup dialog. In this case, the page is still there. We will call for the re-authorization if anything changes and it's certainly possible to re-render this dialog if that's what you want. But this is why I was asking..